src/averageAges.js

```diff
@@ -55,7 +69,27 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const countChilds = people
+    .filter(person => person.mother && people
+      .some(key => key.name === person.mother));
+
+  const arraySons = onlyWithSon
+    ? (countChilds.filter(person => person.sex === 'm'))
+    : countChilds;
+
+  const kidsAge = arraySons
+    .map(person => person.born - people
+      .find(mother => mother.name === person.mother).born);
+
+  const findAgeDiff = kidsAge.reduce((acc, age) => acc + age, 0);
+
+  return findAgeDiff / kidsAge.length;
+}
+
+function getAvargeAge(peopeList) {
```

Comment:
Fix the typo in the `peopeList` parameter name. Change it to `peopleList`.

Reply:
also, function getAvargeAge has a typo and actually it returns sum, not average value